Linux GCC Build Errors: cosf & sinf

### Issue
It seems that when building MinGfx on Linux (specifically I am running Manjaro), the compiler throws errors on the usage of `std::cosf` and `std::sinf`. See the [Error](#error) section below to see the verbose error output. 

### Potential Solution
The solution I would suggest is to use the C++ overloaded operators `std::cos` and `std::sin`, which will use `float` arguments and return values. 

### My Setup

System: Personal Computer
Operating System: Manjaro 21.1.3 "Pahvo"

### Error
**Input:**
`make install -j12`

**Output:**
```
[ 92%] Building CXX object tests/gui_plus_opengl/CMakeFiles/mingfx-test-gui-plus-opengl.dir/gui_plus_opengl.cc.o
cd /home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/build/tests/gui_plus_opengl && /usr/bin/c++ -DLINUX -DUSE_NanoGUI -DUSE_OPENGL -I/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/../../src -I/home/cusp/repos/csci4611/repo-bunze002/include -I/home/cusp/repos/csci4611/repo-bunze002/include/nanovg -I/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/src -fPIC -std=c++14 -MD -MT tests/gui_plus_opengl/CMakeFiles/mingfx-test-gui-plus-opengl.dir/gui_plus_opengl.cc.o -MF CMakeFiles/mingfx-test-gui-plus-opengl.dir/gui_plus_opengl.cc.o.d -o CMakeFiles/mingfx-test-gui-plus-opengl.dir/gui_plus_opengl.cc.o -c /home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc
/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc: In member function ‘virtual void GuiPlusOpenGL::DrawUsingNanoVG(NVGcontext*)’:
/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc:219:38: error: ‘cosf’ is not a member of ‘std’; did you mean ‘cosh’?
  219 |     nvgCircle(ctx, 512.0f+50.0f*std::cosf((float)simTime_), 350.0f+200.0f*std::sinf((float)simTime_), 50.0f);
      |                                      ^~~~
      |                                      cosh
/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc:219:80: error: ‘sinf’ is not a member of ‘std’; did you mean ‘sinh’?
  219 |     nvgCircle(ctx, 512.0f+50.0f*std::cosf((float)simTime_), 350.0f+200.0f*std::sinf((float)simTime_), 50.0f);
      |                                                                                ^~~~
      |                                                                                sinh
/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc:226:39: error: ‘cosf’ is not a member of ‘std’; did you mean ‘cosh’?
  226 |     nvgCircle(ctx, 512.0f+200.0f*std::cosf((float)simTime_), 350.0f+50.0f*std::sinf((float)simTime_), 50.0f);
      |                                       ^~~~
      |                                       cosh
/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/tests/gui_plus_opengl/gui_plus_opengl.cc:226:80: error: ‘sinf’ is not a member of ‘std’; did you mean ‘sinh’?
  226 |     nvgCircle(ctx, 512.0f+200.0f*std::cosf((float)simTime_), 350.0f+50.0f*std::sinf((float)simTime_), 50.0f);
      |                                                                                ^~~~
      |                                                                                sinh
make[2]: *** [tests/gui_plus_opengl/CMakeFiles/mingfx-test-gui-plus-opengl.dir/build.make:76: tests/gui_plus_opengl/CMakeFiles/mingfx-test-gui-plus-opengl.dir/gui_plus_opengl.cc.o] Error 1
make[2]: Leaving directory '/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/build'
make[1]: *** [CMakeFiles/Makefile2:170: tests/gui_plus_opengl/CMakeFiles/mingfx-test-gui-plus-opengl.dir/all] Error 2
make[1]: Leaving directory '/home/cusp/repos/csci4611/repo-bunze002/dev/MinGfx/build'
make: *** [Makefile:136: all] Error 2
```

- Jacob Bunzel (bunze002@umn.edu)

--- a/tests/gui_plus_opengl/gui_plus_opengl.cc
+++ b/tests/gui_plus_opengl/gui_plus_opengl.cc
@@ -216,14 +216,14 @@ void GuiPlusOpenGL::DrawUsingNanoVG(NVGcontext *ctx) {
     // example of drawing some circles
     
     nvgBeginPath(ctx);
-    nvgCircle(ctx, 512.0f+50.0f*std::cos((float)simTime_), 350.0f+200.0f*std::sin((float)simTime_), 50.0f);
+    nvgCircle(ctx, 512.0f+50.0f*GfxMath::cos((float)simTime_), 350.0f+200.0f*GfxMath::sin((float)simTime_), 50.0f);
     nvgFillColor(ctx, nvgRGBA(100,100,255,200));
     nvgFill(ctx);
     nvgStrokeColor(ctx, nvgRGBA(0,0,0,255));
     nvgStroke(ctx);
 
     nvgBeginPath(ctx);
-    nvgCircle(ctx, 512.0f+200.0f*std::cos((float)simTime_), 350.0f+50.0f*std::sin((float)simTime_), 50.0f);
+    nvgCircle(ctx, 512.0f+200.0f*GfxMath::cos((float)simTime_), 350.0f+50.0f*GfxMath::sin((float)simTime_), 50.0f);
     nvgFillColor(ctx, nvgRGBA(255,100,100,200));
     nvgFill(ctx);
     nvgStrokeColor(ctx, nvgRGBA(0,0,0,255));

